src/common/ecs-features/ecsFeatureClient.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import TelemetryReporter from "@vscode/extension-telemetry";
+import { ITelemetry } from "../../client/telemetry/ITelemetry";
+import { getECSRequestURL } from "./ecsFeatureUtil";
+import { Feature } from "./feature";
+
+export abstract class ECSFeaturesClient {
+    private static _ecsConfig: Record<string, string | boolean>;
+    private static _featuresConfig = {};
+
+    // Initialize ECSFeatureClient and reference this for accessing any client config like below
+    // ECSFeaturesClient.getConfig(EnableMultifileVscodeWeb).enableMultifileVscodeWeb
+    public static async init(telemetry: ITelemetry | TelemetryReporter, envId?: string, userId?: string, tenantId?: string, region?: string) {
+        if (!this._ecsConfig) {
+            const requestURL = getECSRequestURL(envId, userId, tenantId, region);
+            try {
+                const response = await fetch(requestURL, {
+                    method: 'GET'
+                });
+                if (!response.ok) {
+                    throw new Error('Request failed');
+                }
+                const result = await response.json();
+                // Update telemetry
+                telemetry.sendTelemetryEvent('ECSConfig', {});
```

Comment:
@ashishchoudhary001 please don't introduce more telemetry

Reply:
@petrochuk didnt get this comment. This is a new feature and we need to collect telemetry here. Without this we are going blind for the copilot features dependent on this change.

Reply:
Sorry, let me clarify. Our team is making progress on moving telemetry to different place with eventual decommission of AppInsights. I am just trying to minimize future work.

Reply:
Raching out offline for this - to help carve out a clearer path ahead. Thank for the prompt response @petrochuk